src/safeposix/syscalls/net_calls.rs

```diff
@@ -631,6 +737,7 @@ impl Cage {
                         );
                     }
 
+                    // ** Why do we not use sock_tmp as the arugment instead of sockfdobj, since sock_tmp is the clone?? ** //
```

Comment:
sock_tmp is the RWLock object that is used for controling read/write access. We need to call  sock_tmp.write() to actually gain the sockhandle object with write lock.